packages/flutter/test/widgets/navigator_replacement_test.dart

```diff
@@ -39,42 +42,202 @@ void main() {
+
+      expect(find.text('home', skipOffstage: false), findsOneWidget);
+      expect(find.text('a', skipOffstage: false), findsOneWidget);
+      expect(find.text('b', skipOffstage: false), findsNothing);
+
+      bool isPushed = false;
+      bool aIsRemoved = false;
+      bool homeIsRemoved = false;
+      observer
+        ..onPushed = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          expectSync(route is PageRoute && route.settings.name == '/b', isTrue);
+          expectSync(previousRoute is PageRoute && previousRoute.settings.name == '/a', isTrue);
+          isPushed = true;
+        }
+        ..onRemoved = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          if (route.settings.name == '/a') {
+            expectSync(previousRoute, null);
+            aIsRemoved = true;
+          } else if (route.settings.name == '/') {
+            expectSync(previousRoute, null);
+            homeIsRemoved = true;
+          }
+        };
+
+      // Remove all routes below
+      navigator.pushNamedAndRemoveUntil('/b', (Route<dynamic> route) => false);
+      await tester.pumpAndSettle();
+
+      expect(find.text('home', skipOffstage: false), findsNothing);
+      expect(find.text('a', skipOffstage: false), findsNothing);
+      expect(find.text('b', skipOffstage: false), findsOneWidget);
+      expect(isPushed, isTrue);
+      expect(aIsRemoved, isTrue);
+      expect(homeIsRemoved, isTrue);
+
+      observer.onPushed = null;
+
+      navigator.pushNamed('/');
+      await tester.pumpAndSettle();
+
+      expect(find.text('home', skipOffstage: false), findsOneWidget);
+      expect(find.text('a', skipOffstage: false), findsNothing);
+      expect(find.text('b', skipOffstage: false), findsOneWidget);
+
+      isPushed = false;
+      homeIsRemoved = false;
+      observer
+        ..onPushed = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          expectSync(route is PageRoute && route.settings.name == '/a', isTrue);
+          expectSync(previousRoute is PageRoute && previousRoute.settings.name == '/', isTrue);
+          isPushed = true;
+        }
+        ..onRemoved = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          expectSync(route is PageRoute && route.settings.name == '/', isTrue);
+          expectSync(previousRoute is PageRoute && previousRoute.settings.name == '/b', isTrue);
+          homeIsRemoved = true;
+        };
+
+      // Remove only some routes below
+      navigator.pushNamedAndRemoveUntil('/a', ModalRoute.withName('/b'));
+      await tester.pumpAndSettle();
+
+      expect(find.text('home', skipOffstage: false), findsNothing);
+      expect(find.text('a', skipOffstage: false), findsOneWidget);
+      expect(find.text('b', skipOffstage: false), findsOneWidget);
+      expect(isPushed, isTrue);
+      expect(homeIsRemoved, isTrue);
+    });
+
+    testWidgets('triggers page transition animation for pushed route', (WidgetTester tester) async {
+
+      observer
```

Comment:
Looks like there's state leaking from one test to another? Either avoid the global state (preferred) or do the clean-up in the test that sets the global state.

Reply:
(I would probably just remove the global myApp and observer and instead instantiate those in the tests directly.